diff --git a/sources/gateway/modules/v1/actions/common/GetMenusAction.php b/sources/gateway/modules/v1/actions/common/GetMenusAction.php
index b6d4eb3..0c030db 100644
--- a/sources/gateway/modules/v1/actions/common/GetMenusAction.php
+++ b/sources/gateway/modules/v1/actions/common/GetMenusAction.php
@@ -37,7 +37,7 @@ class GetMenusAction extends GetApiAction
             $session_user = Yii::$app->userCache->getUser();
 
             $restaurant_id = $requestForm->restaurant_id;
-
+            
             if (!Yii::$app->user->isGuest && isset(Yii::$app->user->identity->restaurant_id)) {
                 $restaurant_id = Yii::$app->user->identity->restaurant_id;
             }
diff --git a/sources/gateway/modules/v1/actions/common/UpdateOrderStatusAction.php b/sources/gateway/modules/v1/actions/common/UpdateOrderStatusAction.php
index 1b9942c..7950c76 100644
--- a/sources/gateway/modules/v1/actions/common/UpdateOrderStatusAction.php
+++ b/sources/gateway/modules/v1/actions/common/UpdateOrderStatusAction.php
@@ -32,7 +32,7 @@ class UpdateOrderStatusAction extends PostApiAction {
         try {
 
             $order = \common\models\Order::find()->where(['id' => $requestForm->order_id])->with('restaurant')->one();
-            $user  = $requestForm->getUser();
+            $user  = Yii::$app->user->identity;
 
             OrderService::changeOrderStatus($order, $requestForm->order_status, $user->user_type, $user->id);
 
diff --git a/sources/gateway/modules/v1/forms/common/LoginForm.php b/sources/gateway/modules/v1/forms/common/LoginForm.php
index 1da4660..ec800bc 100644
--- a/sources/gateway/modules/v1/forms/common/LoginForm.php
+++ b/sources/gateway/modules/v1/forms/common/LoginForm.php
@@ -40,7 +40,13 @@ class LoginForm extends BaseRequestApiForm
     {
         if (!$this->hasErrors()) {
             $this->user = User::findByUsername($this->username);
-            $client_id = $this->getClient()['id'];
+            
+            if (!isset($this->client_key)){
+                $client_id = $this->user->client_id;
+            } else {
+                $client_id = $this->getClient()['id'];
+            }
+            
             if($this->user && $this->user->record_type != RecordType::Active){
                 $this->addError($attribute, Yii::t('error','Account not activated.'));
             }
